tests/kubernetes/bin/test_paasta_secrets_sync.py

```diff
@@ -827,6 +828,81 @@ def crypto_keys_patches():
         )
 
 
+@pytest.fixture
+def database_credentials_patches():
+    with mock.patch(
+        "paasta_tools.kubernetes.bin.paasta_secrets_sync.get_secret_provider",
+        autospec=True,
+    ) as mock_get_secret_provider, mock.patch(
+        "paasta_tools.kubernetes.bin.paasta_secrets_sync.get_secret_signature",
+        autospec=True,
+    ) as mock_get_kubernetes_secret_signature, mock.patch(
+        "paasta_tools.kubernetes.bin.paasta_secrets_sync.create_secret",
+        autospec=True,
+    ) as mock_create_secret, mock.patch(
+        "paasta_tools.kubernetes.bin.paasta_secrets_sync.create_secret_signature",
+        autospec=True,
+    ) as mock_create_kubernetes_secret_signature, mock.patch(
+        "paasta_tools.kubernetes.bin.paasta_secrets_sync.update_secret",
+        autospec=True,
+    ) as mock_update_secret, mock.patch(
+        "paasta_tools.kubernetes.bin.paasta_secrets_sync.update_secret_signature",
+        autospec=True,
+    ) as mock_update_kubernetes_secret_signature, mock.patch(
+        "paasta_tools.kubernetes.bin.paasta_secrets_sync.PaastaServiceConfigLoader",
+        autospec=True,
+    ) as mock_config_loader, mock.patch(
+        "paasta_tools.kubernetes.bin.paasta_secrets_sync.time.sleep",
+        autospec=True,
+    ):
+        yield (
+            mock_get_kubernetes_secret_signature,
+            mock_create_secret,
+            mock_create_kubernetes_secret_signature,
+            mock_update_secret,
+            mock_update_kubernetes_secret_signature,
+            mock_config_loader,
+            mock_config_loader.return_value.instance_configs,
+            mock_get_secret_provider,
+        )
+
+
+def test_sync_database_secrets(database_credentials_patches):
```

Comment:
we probably want to add a couple more tests here (e.g., all the other edge cases like no db credentials configured, db credentials configured but no update required, etc)

additionally, should this test be asserting more than just `sync_database_credentials` returns `True`?

Reply:
++ good point, will add more cases

Reply:
just went through the other sync_* methods - looks like the convention has been to just always return True? any other scenario is just a runtime error :shrug:

Reply:
guess I'll assert the number of times create_or_update_secret is called